cvx/cli/weather_fire.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+import fire
+import requests
+
+
+def cli(metric: str, latitude: float = 37.4419, longitude: float = -122.143) -> None:
+    """
+    Get the current weather for a given metric
+
+    Parameters
+    ----------
+    metric : str
+        The metric to get the current weather for
+    latitude : float, optional
+        The latitude to get the current weather for, by default 37.4419
+    longitude : float, optional
+        The longitude to get the current weather for, by default -122.143
+    """
+    url = "https://api.open-meteo.com/v1/forecast"
+    url = f"{url}?latitude={str(latitude)}&longitude={str(longitude)}&current_weather=true"
+    r = requests.get(url)
+
+    if r.status_code == 200:
+        if metric in r.json()["current_weather"]:
+            x = r.json()["current_weather"][metric]
+            return x
+        else:
+            raise ValueError("Metric not supported!")
+    else:
+        raise ConnectionError("Open-Meteo is down!")
+
+
+def fff():  # pragma: no cover
```

Comment:
I usually put the `Fire` call into a `if __name__ == "__main__` clause.

Reply:
it works better with pyproject.toml if it is hidden in a function which should have a more appropriate name though. I call it main() now... I can then do poetry run weather-fire temperature

Reply:
https://github.com/cvxgrp/cvxmarkowitz/blob/96da221368b7331025f9dc84b3d46bbfa20ca937/pyproject.toml#L53

Reply:
[tool.poetry.scripts]
weather-fire = "cvx.cli.weather_fire:main"

# Relevant documentation:
# https://medium.com/clarityai-engineering/how-to-create-and-distribute-a-minimalist-cli-tool-with-python-poetry-click-and-pipx-c0580af4c026

Reply:
sorry, I didn't try to put any explicit emphasis on the link :-)

Reply:
`[tool.poetry.scripts]
weather-fire = "cvx.cli.weather_fire:main"

# Relevant documentation:
# https://medium.com/clarityai-engineering/how-to-create-and-distribute-a-minimalist-cli-tool-with-python-poetry-click-and-pipx-c0580af4c026
`

Reply:
```
[tool.poetry.scripts]
weather-fire = "cvx.cli.weather_fire:main"

# Relevant documentation:
# https://medium.com/clarityai-engineering/how-to-create-and-distribute-a-minimalist-cli-tool-with-python-poetry-click-and-pipx-c0580af4c026
```